server/src/main/java/com/google/coffeehouse/servlets/JoinClubServlet.java

```diff
@@ -0,0 +1,115 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.coffeehouse.common.MembershipConstants;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to join a club from Http POST Request Body (in JSON format) containing a club
+ * and user ID that exists in the database, and return a success status 200 (OK).
+ */
+@WebServlet("/api/join-club")
+public class JoinClubServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be used to join a club for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** The message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "Body unable to be parsed in JoinClubServlet: ";
+
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a required field.
+   */
+  public static final String NO_FIELD_ERROR = "No \"%s\" found in JSON.";
+  /** The message to be logged when the body of the GET request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in GetProfileServlet: No \"userId\" found in JSON.";
+  /** Name of the key in the input JSON that corresponds to the userId. */
+  public static final String USER_ID_FIELD_NAME = "userId";
+  /** Name of the key in the input JSON that corresponds to the clubId. */
+  public static final String CLUB_ID_FIELD_NAME = "clubId";
+
+  private static final Gson gson = new Gson();
+  private final StorageHandlerApi storageHandler;
+
+  /**
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when adding the membership.
+   */
+  public JoinClubServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = storageHandler;
+  }
+
+  /**
+   * Explicity default constructor used for instantiating the servlet when not testing.
+   */
+  public JoinClubServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+
+  /** 
+   * Adds a membership to the database using user ID and club ID.
+   * @param request the POST request that must have a valid JSON representation of the userId and clubID
+   *     to be passed in order to add a membership to the Memberships table in the database. If this is
+   *     not the case the response will send a "400 Bad Request error"
```

Comment:
Since you describe one of the error cases here, consider adding descriptions of when the other errors are encountered.

Reply:
Thank you for the suggestion! Will definitely be adding that.